File: cpp/ConvertSortedArrayToBinarySearchTree.cpp
#include <vector>
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

class Solution {
public:
    TreeNode *sortedArrayToBST(std::vector<int> &num) {
        int low = 0;
        int high = num.size() - 1;
        int mid = (low + high) / 2;
        if ((int)num.size() == 0) {
            return NULL;
        }
        TreeNode *root = new TreeNode(num[mid]);
        std::vector<int> low_half (num.begin(), num.begin()+mid-low);
        std::vector<int> high_half (num.begin()+mid+1, num.begin()+1+high);
        root->left = this->sortedArrayToBST(low_half);
        root->right = this->sortedArrayToBST(high_half);
        return root;
    }
};

File: java/ConvertSortedArrayToBinarySearchTree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public TreeNode sortedArrayToBST(int[] nums) {
        if (nums.length == 0) {
            return null;
        }
        
        int low = 0;
        int high = nums.length - 1;
        int mid = (low + high) / 2;
        TreeNode root = new TreeNode(nums[mid]);
        root.left = sortedArrayToBST(Arrays.copyOfRange(nums, low, mid));
        root.right = sortedArrayToBST(Arrays.copyOfRange(nums, mid + 1, nums.length));
        return root;
    }
}
